codeclimate/workflows.yaml

```diff
@@ -29,7 +32,11 @@ workflows:
         {{ . }}
         {{- end }}
     steps:
+      - call_function: codeclimate.get_repo_info
       - call_function: codeclimate.add_private_repo
+        with:
+          name: test_reporter_id
+          value: $ctx.repo_info.attributes.test_reporter_id
```

Comment:
Should I conditionally check if repo_info.attributes exist?

Reply:
However, the first step will fail if the repo is not added to codeclimate already, because `get_repo_info` returns a empty list, so the `export` here will fail. https://github.com/honeydipper/honeydipper-config-essentials/blob/e56534558f7a73c6a064f7980f257405e36ee68c/codeclimate/systems.yaml#L101-L102 .

You can change that to below(with a question mark). 
```
        export:
          repo_info: $?data.json.data[0]
```

Reply:
could I try something like this instead? The idea would be to catch the error from the first get repo info call and only call add_private_repo if repo info didn't exist

```
    steps:
      - call_function: codeclimate.get_repo_info
        on_error: continue
        - unless:
          - $?ctx.repo_info
        steps:
          - call_function: codeclimate.add_private_repo
          - call_function: codeclimate.get_repo_info
      - call_function: circleci.add_env_var
        with:
          name: CC_TEST_REPORTER_ID
          value: $ctx.repo_info.attributes.test_reporter_id
          git_repo: "{{ .ctx.org_id }}/{{ .ctx.repo }}"
```